spec/services/category_hashtag_data_source_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+RSpec.describe CategoryHashtagDataSource do
+  fab!(:category1) { Fabricate(:category, slug: "random", topic_count: 432) }
+  fab!(:category2) { Fabricate(:category, slug: "books", topic_count: 235) }
+  fab!(:category3) { Fabricate(:category, slug: "movies", topic_count: 126) }
+  fab!(:group) { Fabricate(:group) }
+  fab!(:category4) { Fabricate(:private_category, slug: "secret", group: group, topic_count: 40) }
+  fab!(:category5) { Fabricate(:category, slug: "casual", topic_count: 99) }
```

Comment:
I think we can improve the fabrication here by avoiding having `topic_count` be in an increasing order. Otherwise, the ordering can be influenced by the `id` of the topic which follows an increasing order as well.